src/mopi/run.py

```diff
@@ -1,17 +1,18 @@
 from typing import List, Tuple
 
-from constants import Const
+from mopi.plugins import WandbConfig, WandbPlugin, OutputAnalyserPlugin
+from mopi.runner.runner import Runner
+from mopi.type import Experiment, StagingConfig, StagingNames
+from mopi.runner.utils import overwrite_preprocessing_configs_
 
-from plugins import WandbConfig, WandbPlugin, OutputAnalyserPlugin
-from runner.runner import Runner
-from type import Experiment, StagingConfig, StagingNames, SourceTypes
-from runner.utils import overwrite_preprocessing_configs_
-from utils.json import dump_str
+from mopi.blocks.io import export_pipeline
 
 
 def run(
     experiments: List[Experiment],
     staging_config: StagingConfig,
+    pure_inference: bool = False,
+    save_entire_pipeline: bool = False,
```

Comment:
This toggle whether we should create a pickle object of the entire pipeline.

⚠️ For some pipelines pickling doesn't work (it can't pickle eg. generators) -> I tried using ``dill`` which extends the functionality of ``pickle`` but it still gave errors.

Reply:
do you want to create a ticket for this - that we should use pickle, but null out `model` before, and then use `load()` on the entire loaded pipeline?

Reply:
Yes, I think that's the way forward.